dashboard/test/factories/workshops.rb

```diff
@@ -126,5 +135,96 @@
     #
     # Sub-factories
     #
+
+    # CSF Workshops, which are usually one-day workshops
+    # that happen year-round.
+    factory :csf_workshop do
+      # Make a CSF 101 "Intro" workshop by default
+      intro
+
+      course Pd::Workshop::COURSE_CSF
+      capacity 30          # Average capacity
+      on_map true          # About 60% are on the map
+      funded               # About 90% are funded
+      num_sessions 1       # Most have 1 session
+      num_facilitators 1   # Most have 1 facilitator
+      each_session_hours 7 # The most common session length
+
+      # CSF Intro, also known as CSF 101
+      # Our most common workshop type as of August 2019.
+      trait :intro do
+        subject Pd::Workshop::SUBJECT_CSF_101
+        location_name 'Walkerville Elementary School'
+      end
+      factory(:csf_intro_workshop, aliases: [:csf_101_workshop]) {intro}
+
+      # CSF Deep Dive, also known as CSF 201
+      trait :deep_dive do
+        subject Pd::Workshop::SUBJECT_CSF_201
+        location_name 'Third Street Elementary School'
+      end
+      factory(:csf_deep_dive_workshop, aliases: [:csf_201_workshop]) {deep_dive}
+    end
+
+    # CSD and CSP Academic Year Workshops
+    # These are one- or two-day workshops on specific parts of our curriculum that happen
+    # throughout the school year.  They have a lot in common.
+    factory :academic_year_workshop do
+      # Make a CSP workshop by default
+      csp
+
+      capacity 30          # Average capacity
+      on_map false         # Never on the map
+      funded               # More than half are funded
+      num_facilitators 2   # Most have 2 facilitators
+
+      # Course can be derived from the chosen subject
+
+      # Workshops 5 and 6 are two-day workshops, others are one-day.
+      num_sessions {TWO_DAY_AYW_SUBJECTS.include?(subject) ? 2 : 1}
```

Comment:
it might be worth adding a trait for a 2-day workshop that automatically gets a valid 2-day subject and 2 sessions - I'm noticing tests that want to create a 2-day workshop and have to either specify the subject (which you probably don't know without looking up), or give the number of sessions and then have the wrong subject...but maybe that's getting too detailed! hard to know where the line is.

Reply:
I thought about this - and since it came to mind for you too, it's probably the right thing to do.  Something like this? `create :csp_academic_year_workshop, :two_day`

Reply:
Done! Let me know what you think. https://github.com/code-dot-org/code-dot-org/pull/30281/commits/03871ade3f6c03c66383872b8901108848509318

Reply:
that's exactly what I was thinking! thanks!